### PR TITLE
Add custom gremlin transport

### LINKED ISSUE #72: Improve logic for retrying client connections on disconnect
Remote Gremlin Server connections can be unexpectedly lost for a variety of reasons. As it currently stands, the `AiohttpTransport` implemented by `gremlinpython` does not really cope well with re-establishing connections. For long-lived Gremlin clients such as the one created by the `nixpkgs-graph-explorer` API this can be especially troublesome since the likelihood of a connection error increases with the longevity of the client.

A possible solution would be to extend the existing `AiohttpTransport` to attempt to reconnect when a failure is detected.

Related StackOverflow: https://stackoverflow.com/questions/70109681/neptune-runtimeerror-connection-was-already-closed-in-pythongremlin

Example traceback (from `etl.py` in this case):

```
Initiating... (removing nodes and edges)
Adding nodes...
Traceback (most recent call last):
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/etl.py", line 187, in <module>
    main()
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/etl.py", line 183, in main
    ingest_graph(df, "ws://10.0.1.3:80/gremlin")
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/etl.py", line 139, in ingest_graph
    unique_insert_node(g, row)
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/etl.py", line 81, in unique_insert_node
    g.V().has("package", "outputPath", str(row["outputPath"])).fold().coalesce(
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/process/traversal.py", line 80, in iterate
    try: self.nextTraverser()
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/process/traversal.py", line 88, in nextTraverser
    return self.next_traverser()
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/process/traversal.py", line 92, in next_traverser
    self.traversal_strategies.apply_strategies(self)
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/process/traversal.py", line 684, in apply_strategies
    traversal_strategy.apply(traversal)
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/driver/remote_connection.py", line 78, in apply
    remote_traversal = self.remote_connection.submit(traversal.bytecode)
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/driver/driver_remote_connection.py", line 105, in submit
    results = result_set.all().result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 440, in result
    return self.__get_result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/driver/resultset.py", line 90, in cb
    f.result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 433, in result
    return self.__get_result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/driver/connection.py", line 89, in _receive
    data = self._transport.read()
  File "/home/dorran_howell/nixpkgs-graph-explorer/etl/venv/lib/python3.9/site-packages/gremlin_python/driver/aiohttp/transport.py", line 106, in read
    raise RuntimeError("Connection was already closed.")
RuntimeError: Connection was already closed.
```

### PR DESCRIPTION
**Pull request by dorranh**
As noted in #72, the default transport implemented by `gremlinpython` lacks a mechanism to reconnect in case of unexpected disconnects, network issues, etc. This PR introduces a custom transport implementation which is built directly on the `websockets` library and includes logic for re-establishing client connections in case of connection errors. 

In addition to the unit tests, I have also manually tested these changes locally and on the demo deployment.

Closes #72

### CODE REVIEW

File: api/explorer/api/gremlin.py [hunk 1]
```diff
@@ -108,17 +108,19 @@ def connect(self, url: str, headers: dict | None = None) -> None:
...
-                additional_headers=headers,
-                open_timeout=self.open_timeout,
-                close_timeout=self.close_timeout,
-                ssl_context=self.ssl_context,
-                logger=self.connection_logger,
-            )
-            self._last_connect_url = url
-            self._last_connect_headers = headers
+        if self._client is not None:
+            self.close()
```
Comment (GuillaumeDesforges) [inline api/explorer/api/gremlin.py:112]: don't we want to prevent a new connetion if the connection has not been closed before?

Comment (dorranh) [reply api/explorer/api/gremlin.py:112]: FYI - the base class implementation is [here](https://github.com/apache/tinkerpop/blob/0075cee46e13ee188ed1a25bb1235a4df2db3a9a/gremlin-python/src/main/python/gremlin_python/driver/transport.py). Unfortunately there is not documentation.

Comment (GuillaumeDesforges) [reply api/explorer/api/gremlin.py:112]: makes sense, could you add a comment with a reference to relevant information? (maybe even just this discussion)

**Review by GuillaumeDesforges — APPROVED** [review]
LGTM 💪

**Pull request merged**